x/trigger/types/genesis.go

```diff
@@ -0,0 +1,104 @@
+package types
+
+import (
+	fmt "fmt"
+
+	types "github.com/cosmos/cosmos-sdk/codec/types"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+)
+
+func NewGenesisState(triggerID, queueStart uint64, triggers []Trigger, gasLimits []GasLimit, queuedTriggers []QueuedTrigger) *GenesisState {
+	return &GenesisState{
+		TriggerId:      triggerID,
+		QueueStart:     queueStart,
+		Triggers:       triggers,
+		GasLimits:      gasLimits,
+		QueuedTriggers: queuedTriggers,
+	}
+}
+
+// DefaultGenesis returns the default trigger genesis state
+func DefaultGenesis() *GenesisState {
+	return NewGenesisState(1, 1, []Trigger{}, []GasLimit{}, []QueuedTrigger{})
```

Comment:
Shouldn't the queue start be 0 here?

In `InitGenesis`, if there aren't any queue entries, it's set to 0.

Also not sure about the trigger id of 1, but I'm not sure if that id is used and then incremented or incremented first. I.e. I'm not sure if it's a count of existing triggers or 1+ that.